### ISSUE TITLE
RFC: Phoneme-Group Force Aligner

### ISSUE THREAD

**Issue by obadx** [obadx/prepare-quran-dataset#47]
## Status
Proposed · Target branch: `fastconformer`

## Purpose
Given one or more audio files and their *transcribed phoneme* scripts, extract a precise **start/end timestamp per phoneme group** (e.g. `قَ`, `اا`, `لَ`) — **not** per individual phoneme (`ا`). Groups are produced by `quran_transcript.chunck_phonemes`. Alignment runs on the **`phonemes` CTC level** only.

## Setup for the developer
```bash
git clone https://github.com/obadx/prepare-quran-dataset  
git checkout fastconformer
uv sync --all-extras        # installs torch, torchaudio, transformers, numpy, librosa, quran-transcript, NeMo, etc.
uv run python -c "import torchaudio, quran_transcript, transformers; print('ok')"
```

Models used:

- `obadx/muaalem-model-v3_2` → `model_type="multi_level_ctc"` (w2v2-bert). Input = feature-extractor features. Phoneme vocab = 43 tokens, CTC blank = `[PAD]` id `0`.
- `obadx/muaalem-fastconformer-base-v1` → `model_type="fastconformer-cache-aware"`. Input = raw audio (`model(raw_audio=..., audio_length=...)`). Phoneme vocab = 44 tokens (`[PAD]`=0, `[EOS]`=1), blank id `0`. Hub repo is gated (401 without token); local checkpoint `results-fastconformer-base-v1/config.json` matches, so `from_pretrained` must support both hub and local paths.

## Public API
`src/prepare_quran_dataset/forced_aligner/forced_aligner.py`
```python
def force_align(
    audio_sources: list[Path | bytes],
    phonemse: list[str],                                    # NB: RFC spelling; use phonemse for parity
    model_type: Literal["w2v2-bert", "fastconformer"],
    model_name_or_path: str,
    batch_size: int = 64,
    return_seconds: bool = False,                           # False → ms (int), True → seconds (float)
    samling_rate: int = 16000,
) -> list[list[tuple[int, int]]] | list[list[tuple[float, float]]]:
```
- `len(audio_sources)` must equal `len(phonemse)`.
- Result: one inner list per audio; one `(start, end)` per phoneme group (order matches `chunck_phonemes`).

## Reference implementation
[`MahmoudAshraf97/ctc-forced-aligner`](https://github.com/MahmoudAshraf97/ctc-forced-aligner) (BSD-2): emissions → `log_softmax` → Viterbi forced alignment (we use `torchaudio.functional.forced_align`, batch==1) → `merge_repeats` → per-group spans with adjacent-blank midpoint padding.

### Constraints
- Align **phonemes level only**; Sifat levels ignored.
- No `<star>` tokens; do **not** append `[EOS]`.
- Package dir: `src/prepare_quran_dataset/forced_aligner/` with `__init__.py` exporting `force_align`.

## Tests
`tests/test_forced_aligner_units.py` — pure helpers only (no model download): group chunking, char-range offsets, segments→spans, frame→time, stride math. Smoke-test `force_align` against `assets/audio-sampels/test.wav` with `obadx/muaalem-model-v3_2`.
Run: `uv run pytest tests/test_forced_aligner_units.py`

## Definition of done
- [ ] Setup on `fastconformer` branch with `uv sync --all-extras`
- [ ] Corrected package path `forced_aligner/forced_aligner.py`
- [ ] Return `list[list[(start,end)]]` per phoneme group per audio (ms ints / s floats)
- [ ] Single-responsibility functions, full typing, docstrings with examples
- [ ] Unit tests pass
